frontend/src/components/custom-tools/prompt-card/PromptCard.jsx

```diff
@@ -411,11 +418,26 @@ function PromptCard({
   };
 
   const handleRunApiRequest = async (docId) => {
+    const runId = generateUUID();
```

Comment:
@tahierhussain There were 2 APIs where this runId may be required. 
1. fetch_response
2. single_pass_extraction
Is this handled in both places?